homeassistant/components/omie/sensor.py

```diff
@@ -0,0 +1,107 @@
+"""Sensor for the OMIE - Spain and Portugal electricity prices integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from zoneinfo import ZoneInfo
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CURRENCY_EURO, UnitOfEnergy
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+from homeassistant.util.dt import utcnow
+
+from .const import CET, DOMAIN
+from .coordinator import OMIECoordinator
+from .util import _pick_series_cet
+
+_LOGGER = logging.getLogger(__name__)
+PARALLEL_UPDATES = 0
+
+_ATTRIBUTION = "Data provided by OMIE.es"
+
+
+@dataclass(frozen=True)
+class OMIEPriceEntityDescription(SensorEntityDescription):
+    """Describes OMIE price entities."""
+
+    def __init__(self, key: str) -> None:
+        """Construct an OMIEPriceEntityDescription that reports prices in €/kWh."""
+        super().__init__(
+            key=key,
+            has_entity_name=True,
+            translation_key=key,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+            icon="mdi:currency-eur",
+            suggested_display_precision=4,
+        )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up OMIE from its config entry."""
+    coordinator: OMIECoordinator = entry.runtime_data
+
+    device_info = DeviceInfo(
+        configuration_url="https://www.omie.es/en/market-results",
+        entry_type=DeviceEntryType.SERVICE,
+        identifiers={(DOMAIN, entry.entry_id)},
+        manufacturer="OMI Group",
+        name="OMIE",
+        model="MIBEL market results",
```

Comment:
```suggestion
        name="OMIE",
```
Manufacturer and model don't seem to apply here